include/kernels/coarse_op_kernel_mma.cuh

```diff
@@ -0,0 +1,281 @@
+        coarse_parity &= 1;
+        coord_coarse[0] /= 2;
+
+        int coarse_x_cb = ((coord_coarse[3] * arg.xc_size[2] + coord_coarse[2]) * arg.xc_size[1] + coord_coarse[1])
+            * (arg.xc_size[0] / 2)
+          + coord_coarse[0];
+
+        auto &tile = arg.vuvTile;
+        // We do coarseColor x coarseColor x fineColor
+
+        constexpr bool a_dagger = true;
+        constexpr bool b_dagger = false;
+
+        constexpr int M = tile.m;
+        constexpr int N = tile.n;
+        constexpr int K = tile.k;
+
+        constexpr int lda = N; // Since a_dagger == true here it's N instead of K.
+        constexpr int ldb = N;
+        constexpr int ldc = N * coarseSpin;
+
+        extern __shared__ half smem_ptr[];
+
+        using Config = MmaConfig<M, N, K, lda, ldb, ldc, bM, bN, bK, block_y, block_z>;
+
+        constexpr int m_offset = 0;
+        constexpr int n_offset = 0;
+
+        static_assert(M <= bM, "Dividing M has NOT been implemented yet.\n");
+        static_assert(N <= bN, "Dividing N has NOT been implemented yet.\n");
+        static_assert(K <= bK, "Dividing K has NOT been implemented yet.\n");
+
+        typename Config::SmemObjA smem_obj_a_real(smem_ptr);
+        typename Config::SmemObjA smem_obj_a_imag(smem_obj_a_real.ptr + Config::smem_lda * bK);
+        typename Config::SmemObjB smem_obj_b_real(smem_obj_a_imag.ptr + Config::smem_lda * bK);
+        typename Config::SmemObjB smem_obj_b_imag(smem_obj_b_real.ptr + Config::smem_ldb * bK);
+
+        WarpRegisterMapping wrm((threadIdx.z * blockDim.y + threadIdx.y) * blockDim.x + threadIdx.x);
+
+        using op_c_type = MmaOperandC<typename Config::accumuate_reg_type>;
+
+        typename Config::ALoader a_loader;
+        typename Config::BLoader b_loader;
+
+        // Not unrolling to lift regiter pressure
+        for (int s = 0; s < fineSpin; s++) {
+
+          auto a = arg.AV.wrap(parity, x_cb, s);
+
+          __syncthreads();
+          a_loader.g2r<Config::lda, a_dagger>(a, m_offset, 0);
+          a_loader.r2s<a_dagger>(smem_obj_a_real, smem_obj_a_imag);
+          __syncthreads();
+
+          for (int s_col = 0; s_col < fineSpin; s_col++) { // which chiral block
+
+            auto b = arg.UV.wrap(parity, x_cb, s_col * fineSpin + s);
+
+            __syncthreads();
+            b_loader.g2r<Config::ldb, b_dagger>(b, n_offset, 0);
+            b_loader.r2s<b_dagger>(smem_obj_b_real, smem_obj_b_imag);
+            __syncthreads();
+
+#pragma unroll 1
+            for (int c = 0; c < Config::warp_cycle; c++) {
+
+              // The logical warp assigned to each part of the matrix.
+              int logical_warp_index = wrm.warp_id * Config::warp_cycle + c;
+              int warp_row = logical_warp_index / Config::tile_col_dim;
+              int warp_col = logical_warp_index - warp_row * Config::tile_col_dim;
+
+              op_c_type op_c_real;
+              op_c_type op_c_imag;
+
+#pragma unroll 1
```

Comment:
Does `#pragma unroll 1` actually do anything?

Reply:
It does nothing: I put it here just to indicate that the loop is not unrolled to reduce register pressure.